tfprotov5/tf5server/server.go

```diff
@@ -1202,3 +1202,56 @@ func invalidDeferredResponseDiag(reason tfprotov5.DeferredReason) *tfprotov5.Dia
 			fmt.Sprintf("Deferred reason - %q", reason.String()),
 	}
 }
+
+func (s *server) ValidateListResourceConfig(ctx context.Context, protoReq *tfplugin5.ValidateListResourceConfig_Request) (*tfplugin5.ValidateListResourceConfig_Response, error) {
+	rpc := "ValidateListResourceConfig"
+	ctx = s.loggingContext(ctx)
+	ctx = logging.RpcContext(ctx, rpc)
+	ctx = logging.ResourceContext(ctx, protoReq.TypeName)
+	ctx = s.stoppableContext(ctx)
+	logging.ProtocolTrace(ctx, "Received request")
+	defer logging.ProtocolTrace(ctx, "Served request")
+
+	req := fromproto.ValidateListResourceConfigRequest(protoReq)
+
+	logging.ProtocolData(ctx, s.protocolDataDir, rpc, "Request", "Config", req.Config)
+
+	ctx = tf5serverlogging.DownstreamRequest(ctx)
+
+	// TODO: Remove this check and error in preference of
+	// s.downstream.ValidateListResourceConfig below once ProviderServer interface
+	// implements this RPC method.
+	// nolint:staticcheck
+	listResourceServer, ok := s.downstream.(tfprotov5.ProviderServerWithListResource)
+	if !ok {
+		logging.ProtocolError(ctx, "ProviderServer does not implement ValidateListResourceConfig")
+
+		protoResp := &tfplugin5.ValidateListResourceConfig_Response{
```

Comment:
A non-blocking thought for follow-up: in light of recent experience with unimplemented RPCs, I'm curious if Terraform prefers the provider to return an error instead of a diagnostic.

https://pkg.go.dev/google.golang.org/grpc/status#Error and https://pkg.go.dev/google.golang.org/grpc@v1.72.2/codes#Code – `Unimplemented Code = 12`.